### ISSUE TITLE
Not truly useable as a tool for developers of reusable actions

### ISSUE THREAD

**Issue by douglascamata** [rhysd/actionlint#350]
I tried using `actionlint` in a small reusable Github Action that I develop and I got a lot of false positives (see below).

```sh
$ actionlint action.yml 
action.yml:1:1: "on" section is missing in workflow [syntax-check]
  |
1 | name: "Setup Docker on macOS"
  | ^~~~~
action.yml:1:1: "jobs" section is missing in workflow [syntax-check]
  |
1 | name: "Setup Docker on macOS"
  | ^~~~~
action.yml:2:1: unexpected key "description" for "workflow" section. expected one of "concurrency", "defaults", "env", "jobs", "name", "on", "permissions", "run-name" [syntax-check]
  |
2 | description: "Setup Docker on macOS using Colima, Lima-VM, and Homebrew."
  | ^~~~~~~~~~~~
action.yml:3:1: unexpected key "inputs" for "workflow" section. expected one of "concurrency", "defaults", "env", "jobs", "name", "on", "permissions", "run-name" [syntax-check]
  |
3 | inputs:
  | ^~~~~~~
action.yml:8:1: unexpected key "outputs" for "workflow" section. expected one of "concurrency", "defaults", "env", "jobs", "name", "on", "permissions", "run-name" [syntax-check]
  |
8 | outputs:
  | ^~~~~~~~
action.yml:18:1: unexpected key "runs" for "workflow" section. expected one of "concurrency", "defaults", "env", "jobs", "name", "on", "permissions", "run-name" [syntax-check]
   |
18 | runs:
   | ^~~~~
```

All these missing sections and unexpected keys are completely fine when you are developing a reusable Github Action that gets published in the marketplace. For reference, I ran the tool over [this file](https://github.com/douglascamata/setup-docker-macos-action/blob/50a8f0817b525f54abf6f06617ebb3c83e92f8d1/action.yml).

I would love if this tool had a toggle to tell it the action being linted is a reusable one and it could adapt accordingly to such scenario. 

Thanks!

**Comment by douglascamata** [comment]
Very likely, that's its current state, yes. But why not cover both? 😄

**Comment by AlexandreDecan** [comment]
Because linting an Action is different than linting a workflow. The name of the tool is maybe a bit misleading, but its purpose is well defined on the homepage: _"[actionlint](https://github.com/rhysd/actionlint) is a static checker for GitHub Actions workflow files."_

**Comment by douglascamata** [comment]
@AlexandreDecan it's not much different (did I ever say they were the same thing?), besides a few schema items. Anyway, this issue is a suggestion/feature request to the project. A project's purpose might change over time. :)

**Comment by AlexandreDecan** [comment]
Indeed :-) But remember you started your message with "I got a lot of false positive", my point was it's not surprising to have false positives if the tool is applied on the wrong file :-D

**Comment by cmotsn** [comment]
This might be a duplicate of https://github.com/rhysd/actionlint/issues/46 ?

**Comment by planeiii-te** [comment]
> Because linting an Action is different than linting a workflow

It would be useful to make this distinction clear in the readme since the name is throwing off lots of people:

> [!IMPORTANT]
>
> `actionlint` does not and will not work as expected for `action.yaml` files. It is only capable of linting files in the `.github/workflows` directory.

.

**Comment by only1thor** [comment]
This is apparently fixed (but not merged) in:
https://github.com/rhysd/actionlint/pull/366
But i have not tried it myself. just skimmed the comments.
